bliss/sleep.py

```diff
@@ -413,6 +423,19 @@ def get_metrics(self, batch):
                 for mse in fluxes_mse:
                     fluxes_mse_vec.append(mse.item())
 
+                # calculate tpr and ppv
+                tpr_i, ppv_i = sdss.get_summary_stats(
```

Comment:
I actually think is better if you calculate these outside of the if statement: `if true_n_sources_i == n_sources_i:` , I think the function should be able to handle locs, fluxes, etc. in batches. I'm not sure about the error you pointed out though, it seems there is some error happening because the locations are empty? Does it work when they are not empty?